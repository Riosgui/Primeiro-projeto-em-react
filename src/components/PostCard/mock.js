export const PropsPostCardMock = {
  title: 'Algo',
  body: 'Que',
  id: 1,
  cover: 'img/quero.png',
};
