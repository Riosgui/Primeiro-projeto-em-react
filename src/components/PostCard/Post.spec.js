import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import { PropsPostCardMock } from "./mock";

const props = PropsPostCardMock

describe('<PostCard />', () => {
    it('should render PostCard correctly', () => {
        render(<PostCard {...props} />);     
        
        expect(screen.getByRole('img', {name: /Algo/i})).toHaveAttribute('src', props.cover);
        expect(screen.getByText('Que')).toBeInTheDocument();
    });

    it('should match snapshot', () => {
        const { container } = render(<PostCard {...props} />);
        // eslint-disable-next-line testing-library/no-node-access
        expect(container.firstChild).toMatchSnapshot();
    });
});