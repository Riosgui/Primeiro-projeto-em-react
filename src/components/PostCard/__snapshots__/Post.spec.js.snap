// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Algo"
    src="img/quero.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Algo
    </h2>
    <p>
      Que
    </p>
  </div>
</div>
`;
